fixtures/flight/server/handler.server.js

```diff
@@ -20,7 +20,8 @@ module.exports = function(req, res) {
         const App = m.default.default || m.default;
         res.setHeader('Access-Control-Allow-Origin', '*');
         const moduleMap = JSON.parse(data);
-        pipeToNodeWritable(React.createElement(App), res, moduleMap);
+        const {pipe} = renderToNodePipe(React.createElement(App), moduleMap);
```

Comment:
Apologies in advance for the unsolicited bikeshedding. The name `renderToNodePipe` just feels odd to me. Thoughts on something like `streamRoot(…).pipe(…)` or `renderStreamingRoot(…).pipe(…)` for symmetry with `hydrateRoot` and `createRoot`?

Reply:
`createRoot` and `hydrateRoot` deal with a "root" object. The object is important because it's stateful (you can update a root or unmount a root later).

This seems a bit different because (1) you don't get a root object — it's a client-only concept, and (2) it's not possible to update it or such. So it seems misleading. I'd expect something like `streamRoot` to accept or produce a "root" object, which it doesn't.

Reply:
My thinking around the “Node” part is that it’s worth clarifying that it’s non-standard. If it was to remain the defacto standard maybe we wouldn’t need the Node prefix but that doesn’t seem to be the case.

Since the modules are auto-swapped by condition we don’t really need a unique name. It can be the same name. However type systems don’t seem to consider the conditional environments yet so it’s nicer to just pretend all the exports exist for now.

Reply:
> createRoot and hydrateRoot deal with a "root" object. The object is important because it's stateful (you can update a root or unmount a root later).

Well, it is (or was) stateful in a sense. You couldn’t update it, but you can tell it to start writing or to abort. And you give it temporary ownership of a stream, just like you do a DOM node.

I guess the bigger issue is that it sounds like “root” might have a different meaning: I assumed it just referred to the root of a React tree that can be controlled over time, but it sounds like it specifically means a root host element?

Reply:
The way I think about “hydrateRoot(…)” is that you’re extracting a root that already existed (conceptually).

So in that sense it seems like someone just have created it at some point on the server.

Reply:
I have some ideas for rendering non-root subtrees for subsequent requests. Those would not be the same as roots on the client. However, they would likely be different APIs. Using the word root might help disambiguate them.